### PR TITLE
Rename confusing names and add some javadoc in n.m.server.world

### LINKED ISSUE #3628: Rename ChunkTicketManager.chunkHolders to make its usage more clear
Normally, a name like `chunkHolders` may sounds like a set of all existing or accessible ChunkHolder instances. However, it only actually contains chunks whose loading levels have been changed between calls on `ChunkTicketManager.tick(TACS storage)`, that is, chunks requiring further loading or generation depending on their updated levels to be performed.

Consequently, names like `chunkHoldersToBeUpdated`, `chunkHoldersWithUpdatedLevels` or `updatedChunkHolders` may be more perferrable than the currently used one.

### PR DESCRIPTION
**Pull request by lovexyn0827**
Fixes: #3628 and #3545

### CODE REVIEW

**Comment by modmuss50** [comment]
Did you mean to make this PR for 23w14a? If not we can get the bot to update it to 23w32a.

**Comment by lovexyn0827** [comment]
> Did you mean to make this PR for 23w14a? If not we can get the bot to update it to 23w32a.

No, updating it to 23w32a will be fine. 
It is only because the default branch of my fork is 23w14a and i forgot to fetch recent changes.

File: mappings/net/minecraft/server/world/ThreadedAnvilChunkStorage.mapping [hunk 1]
```diff
@@ -56,6 +56,8 @@ CLASS net/minecraft/class_3898 net/minecraft/server/world/ThreadedAnvilChunkStor
 	METHOD method_17216 getChunkHolder (J)Lnet/minecraft/class_3193;
 		ARG 1 pos
 	METHOD method_17217 setLevel (JILnet/minecraft/class_3193;I)Lnet/minecraft/class_3193;
+		COMMENT Sets the loading level of {@code ChunkHolders}. Nonexistent {@code ChunkHolder}s will be created automaiccally
```
Comment (apple502j) [inline mappings/net/minecraft/server/world/ThreadedAnvilChunkStorage.mapping]: typo?

Comment (lovexyn0827) [reply mappings/net/minecraft/server/world/ThreadedAnvilChunkStorage.mapping]: yes, sorry for this mistake

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**